src/Orleans.Transactions/DeadlockDetection/DeadlockDetector.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Orleans.Concurrency;
+using Orleans.Runtime;
+using Orleans.Transactions.Abstractions;
+
+namespace Orleans.Transactions.DeadlockDetection
+{
+
+    [Reentrant]
+    public class DeadlockDetector : Grain, IDeadlockDetector
+    {
+        private readonly ILogger<DeadlockDetector> logger;
+        public DeadlockDetector(ILogger<DeadlockDetector> logger)
+        {
+            this.logger = logger;
+        }
+
+        public async Task CheckForDeadlocks(ParticipantId resourceId, IList<Guid> transactionIds)
+        {
+            // Right now, this is called multiple times (once for each deadlocked transaction queue).  To avoid
+            // this, we might want to  keep track of things we're currently examining and do a fast exit if
+            // another call is already on it.
+
+            var txStr = string.Join(",", transactionIds);
+            this.logger.LogInformation($"Checking for deadlocks around {resourceId} for {txStr}");
```

Comment:
I'd like to be using structured logging and I believe the Orleans team wants to move that direction as well